app/controllers/miq_policy_controller.rb

```diff
@@ -547,73 +547,83 @@ def replace_right_cell(nodetype, replace_trees = [], presenter = nil)
+                            when 'compliance' then _("Compliance Policy")
+                            end
           r[:partial => 'policy_folders']
         elsif @alert_profiles
-          right_cell_text = _("All %{typ} %{model}") % {:typ   => ui_lookup(:model => @sb[:folder].try(:camelize)),
-                                                        :model => ui_lookup(:models => 'MiqAlertSet')}
+          right_cell_text = _("All %{towhat} Alert Profiles") % {:towhat => ui_lookup(:model => @sb[:folder].try(:camelize))}
           r[:partial => 'alert_profile_list']
         end
       )
     when 'p'
       presenter.update(:main_div, r[:partial => 'policy_details', :locals => {:read_only => true}])
-      model_name = ui_lookup(:model => @sb[:nodeid].try(:camelize))
+      towhat = ui_lookup(:model => @sb[:nodeid].try(:camelize))
       if @policy.id.blank?
-        right_cell_text = if @sb[:mode]
-                            _("Adding a new %{model_name} %{mode} Policy") %
-                              {:model_name => model_name, :mode => @sb[:mode].capitalize}
-                          else
-                            _("Adding a new %{model_name} Policy") % {:model_name => model_name}
+        right_cell_text = case @sb[:mode]
+                          when 'control'
+                            _("Adding a new %{towhat} Control Policy") % {:towhat => towhat}
+                          when 'compliance'
+                            _("Adding a new %{towhat} Compliance Policy") % {:towhat => towhat}
+                          when nil
+                            _("Adding a new %{towhat} Policy") % {:towhat => towhat}
                           end
       else
-        options = {:model => "#{model_name} #{@sb[:mode] ? @sb[:mode].capitalize : ""} Policy",
-                   :name  => @policy.description.gsub(/'/, "\\'")}
-        right_cell_text = @edit ? _("Editing %{model} \"%{name}\"") % options : _("%{model} \"%{name}\"") % options
-        if @edit && @edit[:typ] == 'conditions'
-          right_cell_text += _(" %{model} Assignments") % {:model => ui_lookup(:model => 'Condition')}
-        end
-        if @edit && @edit[:typ] == 'events'
-          right_cell_text += _(" %{model} Assignments") % {:model => ui_lookup(:model => 'Event')}
-        end
+        options = {:towhat => towhat,
+                   :mode   => case @sb[:mode]
+                              when 'control'    then _("Control")
+                              when 'compliance' then _("Compliance")
+                              when nil          then "" # TODO: when can it be nil?
+                              end,
+                   :name   => @policy.description.gsub(/'/, "\\'")}
+        right_cell_text = if @edit
+                            if @edit[:typ] == 'conditions'
+                              _("Editing %{towhat} %{mode} Policy \"%{name}\" Condition Assignments") % options
```

Comment:
I should turn these %{mode} into 2*4 strings with "Control Policy" / "Condition Policy".
But I don't really want to make it 3*4 strings covering `@sb[:mode]` being nil...